### ISSUE TITLE
Imagic: Text-Based Real Image Editing with Diffusion Models

### ISSUE THREAD

**Issue by nmhkahn** [nmhkahn/paper-reading#38]
## Motivation
- (image, target prompt) 만 가지고 real image editing 하는 task
- 풀고자 하는 문제는 #36 과 동일 (심지어 둘다 구글리서치고 arXiv pub date도 거의 비슷함...)

## Method
<img width="650" alt="Screenshot 2022-11-23 at 2 17 18 PM" src="https://user-images.githubusercontent.com/16869368/203473546-c1e42bad-1c2d-46c8-b01a-4d83263eda91.png">

- 크게 3가지의 프로세스를 거쳐서 target prompt $e_{tgt}$ 기반 editing 진행
1. Text embedding optimization
    - $e_{tgt}$를 pre-trained DM에 넣어서 DM loss를 가지고 $e_{tgt}$를 optimization -> $e_{opt}$를 obtain
    - $e_{opt}$는 input image를 최대한 따라가도록 학습하되, few step만 진행 (initial target prompt를 보존하기 위해)
    - 따라서 $e_{tgt}$와 $e_{opt}$를 interpolation 하면 의미론적인 interpolation이 가능해짐
2. Model fine-tuning
    - $e_{opt}$가 input image를 완전히 recon 할 수 없기 때문에 DM 모델을 약간만 fine-tuning 시켜줌
    - 이 때,  DM loss 그대로 사용하고, $e_{opt}$는 freeze
3. Text embedding interpolation
    - <img width="300" alt="Screenshot 2022-11-23 at 2 23 48 PM" src="https://user-images.githubusercontent.com/16869368/203474184-f3aa721e-9283-4a27-979f-ef370f842d6a.png">
    - $\bar{e}$를 DM에 넣어서 이미지 생성. $\tau$ 값에 따라 얼마나 tgt prompt에 따라갈지 정할 수 있음

**Comment by nmhkahn** [comment]
## Discussion
- "This entire optimization process takes around 8 minutes per image on two TPUv4 chips."
- UniTune과 동일하게 성능 자체는 괜찮은데 거의 실사용이 어려운 연구
- Model fine-tune 단계가 없으면 어느정도 괜찮은 시간이 나올거 같긴한데, 이 단계가 없으면 전체적인 structure가 유지가 잘 안되는 것 같음 (Figure 5)
- 물론, Stable diffusion 처럼 latent 기반 모델로 적용하면 좀 더 나은 latency 기대 가능..?
    - arXiv v2 버전에서 이렇다고 함
    - <img width="350" alt="Screenshot 2022-11-24 at 9 02 26 AM" src="https://user-images.githubusercontent.com/16869368/222627663-d411be95-5661-4c68-9d43-9a6000c14c95.png">
